bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/link/ItemChannelLinkRegistry.java

```diff
@@ -229,4 +270,171 @@ public enum ItemChannelLinkProblem {
+            Class<? extends Tag> type = SemanticTags.getById(tag);
+            if ((type != null) && (Point.class.isAssignableFrom(type) || Property.class.isAssignableFrom(type))) {
+                alreadyHasPointOrPropertyTag = true;
+                break;
+            }
+        }
+
+        Set<String> channelDefaultTags = getChannelDefaultTags(link);
+        if (!channelDefaultTags.isEmpty()) {
+            if (alreadyHasPointOrPropertyTag) {
+                logger.warn("Item '{}' already tagged; forbidden to assign tags from channel '{}'.",
+                        activeItem.getName(), link.getLinkedUID());
+            } else {
+                Set<String> newTags = new HashSet<>(activeItem.getTags());
+                newTags.addAll(channelDefaultTags);
+                logger.info("Item '{}' assigned tags '{}' from channel '{}'.", activeItem.getName(), channelDefaultTags,
+                        link.getLinkedUID());
+
+                link.setTagsLinked(true);
+
+                updateExistingRegistryItemTagsAndNotifyRegistryListeners(activeItem.getName(), newTags);
+            }
+        }
+    }
+
+    private void assignChannelDefaultTags(ItemChannelLink link) {
+        try {
+            if (itemRegistry.getItem(link.getItemName()) instanceof ActiveItem activeItem) {
+                assignChannelDefaultTags(link, activeItem);
+            }
+        } catch (ItemNotFoundException e) {
+        }
+    }
+
+    /**
+     * if the linked channel is the actual source of the item's tags then remove those tags from
+     * the item. If the item had any native custom tags they shall NOT be removed. Finally iterate
+     * over any other linked channels so they may eventually provide new tags.
+     */
+    private void removeChannelDefaultTags(ItemChannelLink oldLink, ActiveItem activeItem) {
+        if (oldLink.tagsLinked()) {
+            Set<String> newTags = new HashSet<>(activeItem.getTags());
+
+            // remove old link's tags
+            Set<String> oldLinkTags = getChannelDefaultTags(oldLink);
+            newTags.removeAll(oldLinkTags);
+            logger.info("Item '{}' removed tags '{}' supplied from channel '{}'.", activeItem.getName(), oldLinkTags,
+                    oldLink.getLinkedUID());
+
+            // iterate over other links in case one may assign new tags
+            boolean alreadyHasPointOrPropertyTag = false;
+            for (ItemChannelLink otherLink : getLinks(activeItem.getName())) {
+                if (!otherLink.getUID().equals(oldLink.getUID())) {
+                    Set<String> otherLinkTags = getChannelDefaultTags(otherLink);
+                    if (!otherLinkTags.isEmpty()) {
+                        if (alreadyHasPointOrPropertyTag) {
+                            logger.warn("Item '{}' already tagged; forbidden to assign tags from channel '{}'.",
+                                    activeItem.getName(), otherLink.getLinkedUID());
```

Comment:
Same here, no need for a warning

Reply:
Same comment as above..

Reply:
Is it now OK? With the conditional check?